### ISSUE TITLE
feature: multi step scenarios / chained scenarios / sub-scenarios

### ISSUE THREAD

**Issue by alainchiasson** [metacloud/molecule#898]
# Issue Type

- Feature request

I'm putting this in as a feature request, but if there are other ways of doing this that I missed, I'm fully open for suggestions. 

# Molecule and Ansible details

```
ansible --version
# ansible 2.3.1.0
#   config file =
#   configured module search path = Default w/o overrides
#  python version = 2.7.10 (default, Feb  7 2017, 00:08:15) [GCC 4.2.1 Compatible Apple LLVM 8.0.0 # (clang-800.0.34)]

molecule --version
# molecule, version 2.0.0.0rc11

```

- Molecule installation method: pip
- Ansible installation method: OS package

# Desired Behaviour

I'm looking for the ability to do multi step testing. As an example:

- Create a VM
- run the installation part of the role
- verify the installation is correct
- run a reconfigure
- Verify the reconfigure is correct

Currently, I can test this with a second scenario that will do both steps. This causes lengthy rebuild cycles. I would like to see the ability to loop through playbook + test if the previous test pass.

If there is another way of doing this - or if I am off the mark in the goals of molecule - let me know.

**Comment by retr0h** [comment]
Its not possible to do unless you use the static driver.  However, that means you would need to manage the instance lifecycle yourself.

This is a great idea, but I don't know when it will get implemented.

**Comment by alainchiasson** [comment]
I also found that the lifecycle management would be the tough part - which is why I thought "sub scenario" as opposed to "chaining" . My main goal in posting this was to get feedback on other ways I may have missed. 

One Item I have seen in test frameworks has been the "before test suite", "After test suite"  and the "before each test", "after each test". 

I'm still going to think through some examples. My background is from Chef and test-kitchen. I now do Ansible and need to beef up my python - so this seemed ideal. I also find test-kitchen ansible a little "coerced into working".

**Comment by retr0h** [comment]
Yeah, so Molecule scopes everything to a scenario.  It may have been taken to an extreme, but all files, configs, generated internal files, test files, etc..   All live within the scenario's directory.

This allows scenarios to be very flexible.  I didn't like the idea of having a single file represent all possible configurations of all scenarios/suites (in kitchen speak).

So changing becomes a little tricky b/c if we end up with a Parent -> Child -> Child scenario relationship, then Molecule needs to use the Parent generated Ansible inventory files so that Ansible can properly connect to the instances.

It's probably doable but have to think it through.

**Comment by retr0h** [comment]
I'm going to close this out.  Implementing this feature feels overly complicated in nature.  For now we have implemented #971, which should give the ability to "prepare" a scenario.
